adlfs/spec.py

```diff
@@ -400,12 +405,17 @@ def __init__(
             self.credential is None
             and self.account_key is None
             and self.sas_token is None
-            and self.client_id is not None
         ):
-            (
-                self.credential,
-                self.sync_credential,
-            ) = self._get_credential_from_service_principal()
+            if self.client_id is not None:
+                (
+                    self.credential,
+                    self.sync_credential,
+                ) = self._get_credential_from_service_principal()
+            else:
+                (
```

Comment:
In this case, if a defaultcredential is returned, then line #513 will have a value for creds, which may attempt to authenticate against a public container.  Can you add an "anon" parameter to handle access to a public container?